File: backend/FlashcardsApi/Models/Flashcard.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace FlashcardsApi.Models;

public class Flashcard
{
    [BsonId]
    [BsonRepresentation(BsonType.String)]
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Question { get; set; } = string.Empty;
    public string Answer { get; set; } = string.Empty;
    public int Score { get; set; } = 0;
    public string DeckId { get; set; } = string.Empty;
    public string Explanation { get; set; } = string.Empty;
    public string Topic { get; set; } = string.Empty;
}

File: frontend/flashcards-ui/src/app/models/flashcard.ts
export interface Flashcard {
    id: string;
    question: string;
    /**
     * Alternate phrasings for the question. The backend always
     * returns this array (possibly empty) so treat it as required
     * on the frontend as well to simplify template bindings.
     */
    questions: string[];
    answer: string;
    score: number;
    deckId: string;
    explanation: string;
    questionImage?: string;
    answerImage?: string;
    explanationImage?: string;
    topic: string;
  }
